_posts/2024-04-29-wait-you-guys-are-writing-unit-tests/wait-you-guys-are-writing-unit-tests.Rmd

```diff
@@ -0,0 +1,139 @@
+---
+title: "Wait, you guys are writing unit tests?"
+description: |
+  A reminder.
+author:
+  - name: Chloe Pou-Prom
+    url: {}
+date: 2024-04-29
+output:
+  distill::distill_article:
+    self_contained: false
+categories:
+    - post-miscellaneous
+draft: true
+preview: unit_tests.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+![](unit_tests.png)
+
+Confession: I rarely write unit tests. 😅 [^1]
+
+[^1]: Yes, I know. They're important and they help find critical failures. Just enter "Are unit tests worth it?" in your favorite search engine, and you'll end up with a mountain ⛰️ of blog posts, StackOverflow replies, Reddit comments, heated discussions, etc. that tell you: *yes, unit tests are worth it*.
+
+That being said, I did recently run into an issue that was identified by a unit test!!
+
+## What happened?
+
+**The situation**: We recently updated Data Virtuality (DV).
+
+> [Data Virtuality (DV)](https://datavirtuality.com/) is our data integration platform which consolidates data from various hospital source systems. It offers many advantages:
+>
```

Comment:
These advantages are interesting, but somewhat abstract; I think I suggested "talk more about what DV is", but actually I'd be most interested in reading something like "DV aggregates data from a bunch of source systems and abstracts behind a single SQL DB interface" or something?  (I know it can be gathered that SQL is how one queries DV from later in the post but it still makes sense to present it here.)